preparation.py

```diff
@@ -42,14 +43,20 @@ def overlay_masks(data_dir, dataset, target_dir, category_ids, erode=0, is_small
                                                                                    distances)
                     mask = add_dropped_objects(mask, mask_eroded)
                 mask_overlayed = np.where(mask, category_nr, mask_overlayed)
+        sizes = get_size_matrix(mask_overlayed)
         distances = clean_distances(distances).astype(np.float16)
         target_filepath = os.path.join(target_dir, dataset, "masks", os.path.splitext(image["file_name"])[0]) + ".png"
-        target_filepath_dist = os.path.join(target_dir, dataset, "distances", image["file_name"][:-4])
-        os.makedirs(os.path.dirname(target_filepath), exist_ok=True)
-        os.makedirs(os.path.dirname(target_filepath_dist), exist_ok=True)
+        target_filepath_dist = os.path.join(target_dir, dataset, "distances", os.path.splitext(image["file_name"])[0])
+        target_filepath_sizes = os.path.join(target_dir, dataset, "sizes", os.path.splitext(image["file_name"])[0])
+        if not dir_created:
```

Comment:
@apyskir if thought the whole point of using `exist_ok=True` was not to have this if